src/rowbinary/de.rs

```diff
@@ -43,71 +88,80 @@ impl<'data> RowBinaryDeserializer<'_, 'data> {
+        #[inline(always)]
+        fn $deser_method<V: Visitor<'data>>(self, visitor: V) -> Result<V::Value> {
+            if Validator::VALIDATION {
+                let mut maybe_enum_validator = self.validator.validate($serde_type);
+                ensure_size(&mut self.input, core::mem::size_of::<$ty>())?;
+                let value = self.input.$reader_method();
+                maybe_enum_validator.validate_identifier::<$ty>(value);
+                visitor.$visitor_method(value)
+            } else {
+                ensure_size(&mut self.input, core::mem::size_of::<$ty>())?;
+                visitor.$visitor_method(self.input.$reader_method())
+            }
+        }
+    };
+}
 
-    impl_num!(u32, deserialize_u32, visit_u32, get_u32_le);
+impl<'data, R: Row, Validator> Deserializer<'data>
+    for &mut RowBinaryDeserializer<'_, 'data, R, Validator>
+where
+    Validator: SchemaValidator<R>,
+{
+    type Error = Error;
 
-    impl_num!(u64, deserialize_u64, visit_u64, get_u64_le);
+    impl_num_or_enum!(i8, deserialize_i8, visit_i8, get_i8, SerdeType::I8);
+    impl_num_or_enum!(i16, deserialize_i16, visit_i16, get_i16_le, SerdeType::I16);
 
-    impl_num!(u128, deserialize_u128, visit_u128, get_u128_le);
+    impl_num!(i32, deserialize_i32, visit_i32, get_i32_le, SerdeType::I32);
+    impl_num!(i64, deserialize_i64, visit_i64, get_i64_le, SerdeType::I64);
+    #[rustfmt::skip]
+    impl_num!(i128, deserialize_i128, visit_i128, get_i128_le, SerdeType::I128);
 
-    impl_num!(f32, deserialize_f32, visit_f32, get_f32_le);
+    impl_num!(u8, deserialize_u8, visit_u8, get_u8, SerdeType::U8);
+    impl_num!(u16, deserialize_u16, visit_u16, get_u16_le, SerdeType::U16);
+    impl_num!(u32, deserialize_u32, visit_u32, get_u32_le, SerdeType::U32);
+    impl_num!(u64, deserialize_u64, visit_u64, get_u64_le, SerdeType::U64);
+    #[rustfmt::skip]
+    impl_num!(u128, deserialize_u128, visit_u128, get_u128_le, SerdeType::U128);
 
-    impl_num!(f64, deserialize_f64, visit_f64, get_f64_le);
+    impl_num!(f32, deserialize_f32, visit_f32, get_f32_le, SerdeType::F32);
+    impl_num!(f64, deserialize_f64, visit_f64, get_f64_le, SerdeType::F64);
 
-    #[inline]
+    #[inline(always)]
     fn deserialize_any<V: Visitor<'data>>(self, _: V) -> Result<V::Value> {
         Err(Error::DeserializeAnyNotSupported)
     }
 
-    #[inline]
+    #[inline(always)]
     fn deserialize_unit<V: Visitor<'data>>(self, visitor: V) -> Result<V::Value> {
         // TODO: revise this.
+        // TODO - skip validation?
         visitor.visit_unit()
     }
 
-    #[inline]
-    fn deserialize_char<V: Visitor<'data>>(self, _: V) -> Result<V::Value> {
-        panic!("character types are unsupported: `char`");
-    }
-
-    #[inline]
+    #[inline(always)]
     fn deserialize_bool<V: Visitor<'data>>(self, visitor: V) -> Result<V::Value> {
+        if Validator::VALIDATION {
```

Comment:
Is it not removed for empty `validate()`?

Reply:
maybe this is not needed indeed, if everything inlines properly..